web/src/main/java/org/hillview/dataStructures/PrivateHeatmap.java

```diff
@@ -46,8 +61,12 @@ private void noiseForBucket(
         }
     }
 
-    private static boolean notConfident(long value, int noise) {
-        return value < noise;
+    public void noiseForRange(int left, int right, int top, int bot,
+                               double scale, double baseVariance, /*out*/Noise result) {
+        List<Pair<Integer, Integer>> xIntervals = dx.kadicDecomposition(left, right, IntervalDecomposition.BRANCHING_FACTOR);
+        List<Pair<Integer, Integer>> yIntervals = dy.kadicDecomposition(top, bot, IntervalDecomposition.BRANCHING_FACTOR);
```

Comment:
Yes, let's not make BRANCHING_FACTOR a constant, it should be a field.

Reply:
Not sure what you mean by this, is there something you want to change here? I don't want it to be a variable that gets passed around; I think it should be a constant on the server.

Reply:
See comment above.